api/filters.go

```diff
@@ -349,6 +354,8 @@ func (api *FilterAPI) updateFilterBlueprint(ctx context.Context, filter *models.
 		}
 	}
 
+	newFilter.UniqueTimestamp = timestamp
```

Comment:
Again i don't think we should be setting the unique timestamp for new filter with the old timestamp (I know this gets reset on insert to datastore but is confusing to read).

The previous timestamp should be passed into UpdateFilter method on datastore package and used in the select method.

If the WithUpdates function in go-ns changed there is a possibility for this to break or if not at very least mislead investigation into our mongo documents when we have real issues.